incubator/hnc/internal/validators/anchor_test.go

```diff
@@ -12,8 +12,8 @@ import (
 
 func TestSubnamespaces(t *testing.T) {
 	// Two roots `a` and `b`, with `c` a subnamespace of `a`, `e` a regular child of `c`,
-	// `f` a subnamespace of `e` and `d` a regular child of `b`
-	f := foresttest.Create("--AbcE")
+	// `f` a subnamespace of `e`, `g` a subnamespace of `a` and `d` a regular child of `b`
+	f := foresttest.Create("--AbcEA")
 	h := &Anchor{Forest: f}
 	f.Get("c").UpdateAllowCascadingDelete(true)
```

Comment:
This test is getting so complicated, that I think each subtest should have its own forest at this point. That means that any one test can have a simpler forest.

Also, maybe split the overall test into TestSubnamespaceCreate and TestSubnamespaceDelete? That lets you take one parameter out of the test, so that the Delete test can look like this:

```
tests := []struct{
  name string
  forest string
  acd string // eg if this is "ab", then namespaces a and b will have allowCascadingDelete set to true
  nm string // no need for cnm and pnm
}{{...}}
```

Reply:
SG. Updated.